src/neo/Network/P2P/Payloads/Block.cs

```diff
@@ -1,69 +1,46 @@
 using Neo.Cryptography;
 using Neo.IO;
 using Neo.IO.Json;
+using Neo.Persistence;
 using System;
-using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 
 namespace Neo.Network.P2P.Payloads
 {
-    public class Block : BlockBase, IInventory, IEquatable<Block>
+    public sealed class Block : IEquatable<Block>, IInventory
     {
-        public const int MaxContentsPerBlock = ushort.MaxValue;
-        public const int MaxTransactionsPerBlock = MaxContentsPerBlock - 1;
+        public const int MaxTransactionsPerBlock = ushort.MaxValue;
 
-        public ConsensusData ConsensusData;
+        public Header Header;
         public Transaction[] Transactions;
 
-        private Header _header = null;
-        public Header Header
-        {
-            get
-            {
-                if (_header == null)
-                {
-                    _header = new Header
-                    {
-                        PrevHash = PrevHash,
-                        MerkleRoot = MerkleRoot,
-                        Timestamp = Timestamp,
-                        Index = Index,
-                        NextConsensus = NextConsensus,
-                        Witness = Witness
-                    };
-                }
-                return _header;
-            }
-        }
+        public UInt256 Hash => Header.Hash;
+        public uint Version => Header.Version;
+        public UInt256 PrevHash => Header.PrevHash;
+        public UInt256 MerkleRoot => Header.MerkleRoot;
+        public ulong Timestamp => Header.Timestamp;
+        public uint Index => Header.Index;
+        public byte PrimaryIndex => Header.PrimaryIndex;
```

Comment:
This simplifies things a lot, but don't we also want to have a nonce in the header? It might be useful to have for contracts.

Reply:
The nonce in the block is an unreliable random number, and contracts should not rely on it.